packages/flutter_tools/lib/src/ios/migrations/remove_framework_link_and_embedding_migration.dart

```diff
@@ -0,0 +1,112 @@
+    }
+
+    bool migrationFailure = false;
+    processFileLines(_xcodeProjectInfoFile, (String line) {
+      // App.framework Frameworks reference.
+      // isa = PBXFrameworksBuildPhase;
+      // files = (
+      //    3B80C3941E831B6300D905FE /* App.framework in Frameworks */,
+      if (line.contains('3B80C3941E831B6300D905FE')) {
+        return null;
+      }
+
+      // App.framework Embed Framework reference (build phase to embed framework).
+      // 3B80C3951E831B6300D905FE /* App.framework in Embed Frameworks */,
+      if (line.contains('3B80C3951E831B6300D905FE')
+          || line.contains('741F496821356857001E2961')) { // Ephemeral add-to-app variant.
+        return null;
+      }
+
+      // App.framework project file reference (seen in Xcode navigator pane).
+      // isa = PBXGroup;
+      // children = (
+      //	 3B80C3931E831B6300D905FE /* App.framework */,
+      if (line.contains('3B80C3931E831B6300D905FE')
+          || line.contains('741F496521356807001E2961')) { // Ephemeral add-to-app variant.
+        return null;
+      }
+
+      // Flutter.framework Frameworks reference.
+      // isa = PBXFrameworksBuildPhase;
+      // files = (
+      //   9705A1C61CF904A100538489 /* Flutter.framework in Frameworks */,
+      if (line.contains('9705A1C61CF904A100538489')) {
+        return null;
+      }
+
+      // Flutter.framework Embed Framework reference (build phase to embed framework).
+      // 9705A1C71CF904A300538489 /* Flutter.framework in Embed Frameworks */,
+      if (line.contains('9705A1C71CF904A300538489')
+          || line.contains('741F496221355F47001E2961')) { // Ephemeral add-to-app variant.
+        return null;
+      }
+
+      // Flutter.framework project file reference (seen in Xcode navigator pane).
+      // isa = PBXGroup;
+      // children = (
+      //	 9740EEBA1CF902C7004384FC /* Flutter.framework */,
+      if (line.contains('9740EEBA1CF902C7004384FC')
+          || line.contains('741F495E21355F27001E2961')) { // Ephemeral add-to-app variant.
+        return null;
+      }
+
+      // Embed and thin frameworks in a script instead of using Xcode's link / embed build phases.
+      const String thinBinaryScript = 'xcode_backend.sh\\" thin';
+      if (line.contains(thinBinaryScript)) {
+        return line.replaceFirst(thinBinaryScript, 'xcode_backend.sh\\" embed_and_thin');
+      }
+
+      if (line.contains('/* App.framework ') || line.contains('/* Flutter.framework ')) {
+        migrationFailure = true;
+      }
+
+      return line;
+    });
+
+    if (migrationFailure) {
+      // Print scary message if the user is on Xcode 11.4 or greater, or if Xcode isn't installed.
+      final bool xcodeIsInstalled = _xcode.isInstalled;
+      if(!xcodeIsInstalled || (_xcode.majorVersion > 11 || (_xcode.majorVersion == 11 && _xcode.minorVersion >= 4))) {
+        logger.printError('Your Xcode project requires migration. See https://github.com/flutter/flutter/issues/50568 for details.');
```

Comment:
Migration fails in `complex_layout` since it was created pre 1.0 and was manually migrated with https://github.com/flutter/flutter/pull/8971.
```
complex_layout$ flutter build ios
Building flutter tool...
Running "flutter pub get" in complex_layout...                      0.5s
Building com.yourcompany.complexLayout for device (ios-release)...
Upgrading project.pbxproj
Your Xcode project requires migration. See https://github.com/flutter/flutter/issues/50568 for details.
Encountered error while building for device.
```

Reply:
Consider adding an analytics event for a failed migration.

Reply:
https://github.com/flutter/flutter/issues/51817